cpp/src/decisiontree/batched-levelalgo/quantile.cuh

```diff
@@ -48,59 +44,54 @@ __global__ void computeQuantilesSorted(T* quantiles,
 }
 
 template <typename T>
-void computeQuantiles(T* quantiles,
-                      int n_bins,
-                      const T* data,
-                      int n_rows,
-                      int n_cols,
-                      const std::shared_ptr<raft::mr::device::allocator> device_allocator,
-                      cudaStream_t stream)
+std::shared_ptr<MLCommon::device_buffer<T>> computeQuantiles(
+  int n_bins, const T* data, int n_rows, int n_cols, const raft::handle_t& handle)
 {
-  thrust::fill(
-    thrust::cuda::par(*device_allocator).on(stream), quantiles, quantiles + n_bins * n_cols, 0.0);
+  auto quantiles = std::make_shared<MLCommon::device_buffer<T>>(
```

Comment:
Is `device_buffer` initialize all the elements to 0? Otherwise in scenarios where samples are less than `n_bins`, `quantiles` will have garbage values.